app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/WidgetListConfigureActivity.kt

```diff
@@ -88,6 +104,60 @@ class WidgetListConfigureActivity : SimpleActivity() {
         finish()
     }
 
+    private fun calculatePeriod(): Int {
+        return if (selectedPeriodOption == EVENT_PERIOD_CUSTOM && selectedPeriodValue != null) {
+            when (selectedPeriodValueType) {
+                R.id.dialog_radio_days -> selectedPeriodValue!! * DAY
+                R.id.dialog_radio_weeks -> selectedPeriodValue!! * WEEK
+                else -> selectedPeriodValue!! * MONTH
+            }
+        } else {
+            selectedPeriodOption
+        }
+    }
+
+    private fun showPeriodSelector() {
+        hideKeyboard()
+
+        val items = ArrayList<RadioItem>()
+        items.add(RadioItem(EVENT_PERIOD_ONE_YEAR, getString(R.string.within_the_next_one_year)))
+        items.add(RadioItem(EVENT_PERIOD_TODAY, getString(R.string.today_only)))
+        items.add(RadioItem(EVENT_PERIOD_CUSTOM, getString(R.string.within_the_next)))
```

Comment:
lets add a couple more predefined values, like 1 week, 1 month, 1 year. Sort them like this too, make Today the first one and Custom the last one.

Reply:
I mean predefined ones so that you dont even have to click on Custom, but they will be above it

Reply:
I did it, it's added in a line `items.addAll(getCustomRadioItems())`. Please see screenshot I've added after last commit.

Reply:
ah right, this is done in a really weird way though.. You add some manually, some via the new function. Make it consistent please

Reply:
I did it like this, because I'm reusing these values later in the `updateSelectedPeriod` to get text easily without a need to recalculate everything again.

Reply:
it is still pretty bad, just do it like in other apps please. You dont need to recalculate anything.